coalib/parsing/filters/FilterBase.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+
+
+def overrides(interface_class):
+    def overrider(method):
+        assert(method.__name__ in dir(interface_class))
+        return method
+    return overrider
```

Comment:
Hm that makes me remember java :D Usually not bad this kind of contract programming, but I think we should discuss that in a separate issue if we really want that. Until then I wouldn't use this as this is inconsistent with the rest of the codebase^^